### DISCUSSION TITLE
How to actually use the weights from train.py, without having to use detect.py

### DISCUSSION THREAD

**Discussion by mnj** [ultralytics/yolov5#4620] [Q&A]
Hey,

Read the getting started guides etc, and got to creating a custom dataset to detect some specific object in a video (mp4), I cloned the repo, found lots of images, annotated them in yolov5 format, and used the train.py script and got a pretty good weight? file (the one in \yolov5\runs\train\yolov5s_resultsxxx\weights\best.pt), that was pretty accurate when using detect.py on a mp4 file.

I looked at countless blogs, and videos, everyone just shows how to use detect.py, not actual custom python code that uses the files generated by train.py, I even tried to dissect the detect.py file, but it's really a terrible mess to understand for a torch beginner, does anyone know an example of how to load that best.pt file that was created from train.py, basically I want to run the detection on live video, frame by frame, so can't use detect.py, the program needs to work on the bounding box data  that is detected, but there doesn't seem to be any examples of actual use, outside the simple train.py -> detect.py, usually done in codelab, which isn't practical to real world use, really confused how this is supposed to work.

I read the parts about saving/loading state etc, but isn't that what train.py already did for the model, when i try to load best.py that was generated from train.py it just complains it's not a valid file to load (despite it working fine with detect.py).

Hope someone can point me in the right direction, on how to use the weights/model that was generated from the yolo5 stuff, the docs just make no sense to me, on how to move that trained data into a real world standalone script/app, outside the provided scripts/codelabs.

**Comment by glenn-jocher** [comment]
@mnj see YOLOv5 **PyTorch Hub** inference tutorial for python inference:

## YOLOv5 Tutorials

* [Train Custom Data](https://github.com/ultralytics/yolov5/wiki/Train-Custom-Data)&nbsp; 🚀 RECOMMENDED
* [Tips for Best Training Results](https://github.com/ultralytics/yolov5/wiki/Tips-for-Best-Training-Results)&nbsp; ☘️ RECOMMENDED
* [Weights & Biases Logging](https://github.com/ultralytics/yolov5/issues/1289)&nbsp; 🌟 NEW
* [Supervisely Ecosystem](https://github.com/ultralytics/yolov5/issues/2518)&nbsp; 🌟 NEW
* [Multi-GPU Training](https://github.com/ultralytics/yolov5/issues/475)
* [PyTorch Hub](https://github.com/ultralytics/yolov5/issues/36)&nbsp; ⭐ NEW
* [TorchScript, ONNX, CoreML Export](https://github.com/ultralytics/yolov5/issues/251) 🚀
* [Test-Time Augmentation (TTA)](https://github.com/ultralytics/yolov5/issues/303)
* [Model Ensembling](https://github.com/ultralytics/yolov5/issues/318)
* [Model Pruning/Sparsity](https://github.com/ultralytics/yolov5/issues/304)
* [Hyperparameter Evolution](https://github.com/ultralytics/yolov5/issues/607)
* [Transfer Learning with Frozen Layers](https://github.com/ultralytics/yolov5/issues/1314)&nbsp; ⭐ NEW
* [TensorRT Deployment](https://github.com/wang-xinyu/tensorrtx)

**Comment by FrancisGoncalves** [comment]
Did you ever sort this out?

Thanks
